tests/kani/Array/repeat_const_generic.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Ensure Kani can evaluate `[e; N]` when `N` is a const generic
+
+struct Foo<const N: usize> {
+    field: [u64; N],
+}
+
+impl<const N: usize> Foo<N> {
+    fn new() -> Self {
+        Self { field: [0; N] }
+    }
+}
+
+#[cfg(kani)]
+mod proofs {
+    use super::*;
+
+    #[kani::proof]
+    fn kani_crash() {
+        Foo::<32>::new();
```

Comment:
Perhaps add an assertion that checks something about the returned value? e.g.:
```rust
let f = Foo::<32>::new();
assert_eq!(f.len(), 32);
```
where the `len` method returns `N`?

Reply:
The other file in this directory, `array_init.rs` already contains many tests for the contents of the array. IMO it'd be redundant

Reply:
Although I guess this is const generic now, ok I'll add one.